tools/resample/resample.hpp

```diff
@@ -0,0 +1,170 @@
+
+#include "misc.hpp"
+#include "basicio.hpp"
+
+using namespace NRG;
+
+namespace NRG::Resample{
+
+template <typename T>
+class Resample
+{
+    private:
+    
+        // Dump additional information to stdout?
+        bool verbose = false; // enable with -v
+        std::string inputfn;  // Filename for input data
+        std::string gridfn;   // Filename for a new X grid
+        std::string outputfn; // Filename for resampled data. May be the same as gridfn.
+        int output_precision = 16; // number of digits of precision in the output
+        // const size_t limit = 1000;
+        // const T EPSABS = 1e-12; // numeric integration epsilon (absolute)
+        // const T EPSREL = 1e-8;  // numeric integration epsilon (relative)
+
+        int len;           // number of data points
+        T Xmin, Xmax; // the interval boundaries
+        std::vector<T>  Xpts, Ypts;
+        
+        gsl_interp_accel *acc;
+        gsl_spline *spline;
+        // gsl_integration_workspace *w;
+
+
+
+        void usage() 
+        {
+            std::cout << "\nUsage: resample [-h] [-v] <input> <grid> <output>" << std::endl;
+            std::cout << "-v: toggle verbose messages (now=" << verbose << ")" << std::endl;
+        }
+
+        void parse_param(int argc, char *argv[]) 
+        {
+            char c;
+            while (c = getopt(argc, argv, "hvp:"), c != -1) 
+            {
+                switch (c) 
+                {
+                case 'h':
+                    usage();
+                    exit(EXIT_SUCCESS);
+                case 'v': verbose = true; break;
+                case 'p': output_precision = atoi(optarg); break;
+                default: throw std::runtime_error("Unknown argument " + c);
+                }
+            }
+            int remaining = argc - optind;
+            if (remaining != 3) {
+                about();
+                usage();
+                exit(1);
+            }
+            inputfn  = std::string(argv[optind++]);
+            gridfn   = std::string(argv[optind++]);
+            outputfn = std::string(argv[optind++]);
+        }
+
+        void about() 
+        {
+            std::cout << "Resampling tool" << std::endl;
+            #ifdef __TIMESTAMP__
+            std::cout << "Timestamp: " << __TIMESTAMP__ << std::endl;
+            #endif
+            std::cout << "Compiled on " << __DATE__ << " at " << __TIME__ << std::endl;
+        }
+             
+
+    public:
+        Resample(int argv, char *argc[])
+        {
+            parse_param(argv, argc);
+            if (verbose) about();
+        }
+
+        Resample(std::string inputfn, std::string gridfn, std::string outputfn, bool verbose = false, int output_precision = 16){
+            this->inputfn = inputfn;
+            this->gridfn = gridfn;
+            this->outputfn = outputfn;
+            this->verbose = verbose;
+            this->output_precision = output_precision;
+            if (verbose) about();
```

Comment:
morda about() tu ni potreben, ker bomo verjetno to funkcijo klicali iz nekega drugega programa.